trilium-note/docker-compose.yml

```diff
@@ -0,0 +1,19 @@
+version: "3.7"
+
+services:
+  server:
+    image: zadam/trilium:latest
+    restart: on-failure
+    hostname: "${DEVICE_HOSTNAME}"
+    environment:
+      - PUID=1000
+      - PGID=1000
+      - VERSION=docker
+      - TRILIUM_DATA_DIR=/home/node/trilium-data
+    ports:
+      - 8080:8080
+    volumes:
+      - ${UMBREL_ROOT}/data/storage/trilium:/home/node/trilium-data
```

Comment:
Is there some reason to use Umbrel's shared storage folder as opposed to having the storage be part of the app install? For example: https://github.com/getumbrel/umbrel-apps/blob/8b7c1ce80ba516bea973434c77adddc6c94d2cc6/code-server/docker-compose.yml#L14

Reply:
Yes. Upon stopping and restarting the app the app data was getting removed and also all the data with it. This method fixed that issue.

Reply:
Stopping/restarting an app won't remove the app's data. It only happens if you uninstall the app. To me, this looks like it should be using the location defined using `$APP_DATA_DIR` which is the app install directory.

Reply:
Ok. Will try and see maybe it I was doing something in a worng way.

Reply:
I have double-checked it now. The volumes part has to remain that only because if that is in the app directory after stopping and starting umbrel the app data is getting reset which is a big issue. Keeping it in Umbrel Root storage keeps the user data safe and reliable for long-term use. I don't know if this issue is for this particular app of something else but that is the case for now.

Reply:
I am unable to replicate this. I've set the volume to: `- ${APP_DATA_DIR}/data:/data` and it appears to be working as expected. I can restart the app and all the data is persisted within the app's install directory.